File: legal-intake-backend/src/controllers/authController.js
const authService = require("../services/auth.service");

exports.login = async (request, reply) => {
  const { email, password } = request.body;
  const result = await authService.login(email, password, reply);
  return reply.send(result);
};

exports.signup = async (request, reply) => {
  const { name, email, password, role } = request.body;
  const result = await authService.signup(name, email, password, role);
  return reply.code(201).send(result);
};

exports.requestOTP = async (request, reply) => {
  const { email } = request.body;
  const result = await authService.requestOTP(email);
  return reply.send(result);
};

exports.verifyOTP = async (request, reply) => {
  const { email, otp } = request.body;
  const result = await authService.verifyOTP(email, otp);
  return reply.send(result);
};

exports.resetPassword = async (request, reply) => {
  const { email, newPassword } = request.body;
  const result = await authService.resetPassword(email, newPassword);
  return reply.send(result);
};

File: legal-intake-backend-spring-COMPLETE/legal-intake-backend-spring/src/main/java/com/deepti/legalintake/controller/AuthController.java
package com.deepti.legalintake.controller;

import com.deepti.legalintake.dto.request.*;
import com.deepti.legalintake.service.AuthService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

/**
 * AUTH CONTROLLER - replaces authController.js + authRoutes.js combined
 *
 * In Node (Fastify) routes and handlers were separate files.
 * In Spring Boot they're combined: @RestController class = routes + handlers together.
 *
 * @RestController    = handles HTTP requests, returns JSON automatically
 * @RequestMapping    = base URL prefix for all routes in this class → "/api/auth"
 * @RequiredArgsConstructor = Lombok injects AuthService via constructor
 *
 * @Valid on @RequestBody = trigger Spring validation (@NotBlank, @Email etc on DTOs)
 *   If invalid → GlobalExceptionHandler.handleValidationErrors() returns 400 automatically
 *
 * @Tag = Swagger group label (shows in swagger-ui.html)
 * @Operation = Swagger endpoint description
 */
@RestController
@RequestMapping("/api/auth")
@RequiredArgsConstructor
@Tag(name = "Authentication", description = "Signup, Login, OTP, Password Reset")
public class AuthController {

    private final AuthService authService;

    /** POST /api/auth/login - replaces fastify.post("/login", loginSchema, authController.login) */
    @PostMapping("/login")
    @Operation(summary = "Login", description = "Returns JWT token on success")
    public ResponseEntity<Map<String, Object>> login(@Valid @RequestBody LoginRequest req) {
        return ResponseEntity.ok(authService.login(req));
    }

    /** POST /api/auth/signup */
    @PostMapping("/signup")
    @Operation(summary = "Signup", description = "Create a new user account")
    public ResponseEntity<Map<String, Object>> signup(@Valid @RequestBody SignupRequest req) {
        return ResponseEntity.status(HttpStatus.CREATED).body(authService.signup(req));
    }

    /** POST /api/auth/request-otp */
    @PostMapping("/request-otp")
    @Operation(summary = "Request OTP", description = "Send OTP to email for password reset")
    public ResponseEntity<Map<String, String>> requestOtp(@Valid @RequestBody RequestOtpRequest req) {
        return ResponseEntity.ok(authService.requestOTP(req));
    }

    /** POST /api/auth/verify-otp */
    @PostMapping("/verify-otp")
    @Operation(summary = "Verify OTP")
    public ResponseEntity<Map<String, String>> verifyOtp(@Valid @RequestBody VerifyOtpRequest req) {
        return ResponseEntity.ok(authService.verifyOTP(req));
    }

    /** POST /api/auth/reset-password */
    @PostMapping("/reset-password")
    @Operation(summary = "Reset Password")
    public ResponseEntity<Map<String, String>> resetPassword(@Valid @RequestBody ResetPasswordRequest req) {
        return ResponseEntity.ok(authService.resetPassword(req));
    }
}